### PR TITLE
Phase 1: Upgrade JS Checkout to Modular (Part 1/2)

### PR DESCRIPTION
**Pull request by thisisayush**
References #283 

# Part 1 of JS Modular Checkout:
- Updates to use General Components instead of custom classes, unless necessary.
- Uses the new Modular UI Design
- Font Icons Added

#### JS Checkout
image
#### No-JS Checkout
image

#### JS Checkout
image

#### No-JS Checkout
image

#### JS Checkout
image

#### No JS Checkout
image

#### JS Checkout
image

#### No JS Checkout
image

# Part 2: Make Refresh via AJAX

To be Done

### CODE REVIEW

File: templates/blockonomics_checkout.php [hunk 1]
```diff
@@ -36,69 +39,90 @@
...
+            </table>
+            <table>
+                <tr>
+                    <th>
+                        <!-- Order Address -->
+                        <strong class="bnomics-address-text">To Pay, send <?=__(strtolower($crypto['name']), 'blockonomics-bitcoin-payments')?> at this address:</strong>
+                        <strong class="bnomics-copy-address-text"><?=__('Copied to clipboard', 'blockonomics-bitcoin-payments')?></strong>
+                        <div class="bnomics-copy-container">
+                            <input type="text" value="<?php echo $order['address']; ?>" id="bnomics-address-input" readonly/>
+                            <a href="#" id="bnomics-address-copy" class="blockonomics-icon">
```
Comment (DarrenWestwood) [inline templates/blockonomics_checkout.php:69]: Surrounding the icons in links is not required and can be removed as this adds unexpected behaviour with some themes due to additional css such as :focus, :clicked etc.
For example, the focus styling remains after the qr code has been clicked.
Focus Issue
The same applies to the Refresh Now `<button>`, as wrapping the button with a link should not be required.

**Review by DarrenWestwood — COMMENTED** [review]
The modular design is looking good.

The .png icons can be removed from the PR. Let's also only load the .woff font file and remove the other formats, as most modern browser support .woff. The blockonomics-icons font instructions have been updated accordingly.

**Comment by shivaenigma** [comment]
Note that now we have an additional complexity of changing *bitcoin* to *bitcoin cash* in case of BCH checkout
What I recommended is to completely remove *bitcoin* and move the *6000USD* to the Order# row https://nimb.ws/twfA9Z This will give more vertical space and things will look compact. Feel free to think about this

**Comment by thisisayush** [comment]
@shivaenigma Right now it's not hardcoded, a styling is applied to the "name" of the crypto chosen: https://github.com/blockonomics/woocommerce-plugin/pull/293/files#diff-8d81ebce00651fad5f0a421bc7d60d514e8ca6bd12cc9fc434ab5e46dab827ecR52

**Comment by blockonomics** [comment]
>What I recommended is to completely remove the bitcoin row and move the 6000USD to the Order# row https://nimb.ws/twfA9Z This will give more vertical space and things will look compact. Feel free to think about this

@thisisayush  let us do this. I will this will make the UI more compact and look better

**Comment by thisisayush** [comment]
image


How it looks with Moving the Order ID and Amount in the same row.
One thing to look for: Handling large order IDs as mentioned by @DarrenWestwood

**Review by DarrenWestwood — COMMENTED** [review]
Overall looks good. Some minor comments

File: templates/blockonomics_checkout.php [hunk 2]
```diff
@@ -36,69 +28,77 @@
...
+            <table>
+                <tr>
+                    <th>
+                        <label class="bnomics-amount-text"><?=__('Amount of', 'blockonomics-bitcoin-payments')?> <?php echo strtolower($crypto['name']); ?> (<?php echo strtoupper($crypto['code']); ?>) <?=__('to send:', 'blockonomics-bitcoin-payments')?></label>
+                        <label class="bnomics-copy-amount-text"><?=__('Copied to clipboard', 'blockonomics-bitcoin-payments')?></label>
 
-                            <span class="bnomics-time-left">00:00 min</span>
+                        <div class="bnomics-copy-container">
+                            <input type="text" value="<?php echo $order_amount; ?>" id="bnomics-amount-input" readonly/>
+                            <a href="#" id="bnomics-amount-copy" class="blockonomics-icon">
```
[the remark this answers is not in the document]
Comment (thisisayush) [reply templates/blockonomics_checkout.php]: Done

File: templates/blockonomics_checkout.php [hunk 3]
```diff
@@ -36,69 +28,77 @@
...
+                            <?php echo $order['value'] ?> <?php echo $order['currency'] ?>
+                        </div>
+                    </th>
+                </tr>
+            </table>
+            <table>
+                <tr>
+                    <th>
+                        <!-- Order Address -->
+                        <label class="bnomics-address-text"><?=__('To Pay, send', 'blockonomics-bitcoin-payments')?> <?php echo strtolower($crypto['name']); ?> <?=__('at this address:', 'blockonomics-bitcoin-payments')?></label>
```
[the remark this answers is not in the document]
Comment (thisisayush) [reply templates/blockonomics_checkout.php]: Done

File: css/order.css [hunk 4]
```diff
@@ -264,3 +260,127 @@ svg {
...
+  padding-right: 5px;
+}
+
+#blockonomics_checkout .bnomics-header {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
+
+#blockonomics_checkout .bnomics-header strong {
```
[the remark this answers is not in the document]
Comment (thisisayush) [reply css/order.css]: DOne

**Review by DarrenWestwood — APPROVED** [review]
Looking good with most themes, although I came across two themes with minor issues.
Hestia: Missing top padding to account for themes with floating headers
Hestia

Astra: Width slightly changes with the time remaining

https://user-images.githubusercontent.com/39666372/199455030-ac01f8d5-5994-4465-b4f1-4cfbacbb0972.mov

**Comment by DarrenWestwood** [comment]
Full pull request with all changes: https://github.com/blockonomics/woocommerce-plugin/pull/301

**Pull request closed without merging**